File: onyx-database-tests/src/main/java/entities/ValidationEntity.java
package entities;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.Attribute;
import com.onyx.persistence.annotations.Entity;
import com.onyx.persistence.annotations.Identifier;

/**
 * Created by timothy.osborn on 1/23/15.
 */
@Entity
public class ValidationEntity extends AbstractEntity implements IManagedEntity
{
    @Attribute
    @Identifier
    public Long id;

    @Attribute(nullable = false)
    public String requiredString;

    @Attribute(size = 10)
    public String maxSizeString;

}

File: onyx-database-tests/src/main/java/entities/ValidationEntity.kt
package entities

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.Attribute
import com.onyx.persistence.annotations.Entity
import com.onyx.persistence.annotations.Identifier

/**
 * Created by timothy.osborn on 1/23/15.
 */
@Entity
class ValidationEntity : AbstractEntity(), IManagedEntity {
    @Attribute
    @Identifier
    var id: Long? = null

    @Attribute(nullable = false)
    var requiredString: String? = null

    @Attribute(size = 10)
    var maxSizeString: String? = null

}
